pkg/component/extensions/operatingsystemconfig/operatingsystemconfig.go

```diff
@@ -979,6 +988,14 @@ func (d *deployer) deploy(ctx context.Context, operation string) (extensionsv1al
 	return d.osc, err
 }
 
+func (o *operatingSystemConfig) calculateKey(version int, worker *gardencorev1beta1.Worker) (string, error) {
+	if v1beta1helper.IsUpdateStrategyInPlace(worker.UpdateStrategy) {
+		return fmt.Sprintf("gardener-node-agent-%s-%s", worker.Name, utils.ComputeSHA256Hex([]byte(worker.Name))[:16]), nil
```

Comment:
Sorry, but why do we need this `utils.ComputeSHA256Hex([]byte(worker.Name))[:16]` part?

Reply:
This is to align, with how the secret hashes look like.

Reply:
Is this cosmetic, or is there a technical reason for it? If no, let's remove it. :)